panic: non-positive interval for NewTicker

Hi, devs! I found that when trying to run the kube-controller-manager binary with the --log-flush-frequency flag with an incorrect value, it panics. For example, this is reproduced with --log-flush-frequency=-0. This is actual for kube-apiserver and kube-scheduler too.

<img width="1974" height="694" alt="Image" src="https://github.com/user-attachments/assets/e69932db-ca0a-40ef-bca0-b1b1bd7a8ab9" />

I suppose that this error should probably be handled separately to avoid causing panic.

<details>

```console
$ go version
go version go1.23.2 linux/amd64

$ sudo KUBERNETES_SERVICE_HOST=localhost KUBERNETES_SERVICE_PORT=8443 ./kube-controller-manager --log-flush-frequency=-0
panic: non-positive interval for NewTicker

goroutine 1 [running]:
time.NewTicker(0xc0002996d0?)
        /usr/local/go/src/time/tick.go:38 +0xbb
k8s.io/klog/v2/internal/clock.RealClock.NewTicker(...)
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/vendor/k8s.io/klog/v2/internal/clock/clock.go:111
k8s.io/klog/v2.(*flushDaemon).run(0xc00035a030, 0x0)
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/vendor/k8s.io/klog/v2/klog.go:1152 +0x106
k8s.io/klog/v2.StartFlushDaemon(0x0)
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/vendor/k8s.io/klog/v2/klog.go:1203 +0x36
k8s.io/component-base/logs/api/v1.apply(0xc0005b5280, 0x0, {0x7056a5bb9458, 0xc0005c4b40})
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/vendor/k8s.io/component-base/logs/api/v1/options.go:265 +0x870
k8s.io/component-base/logs/api/v1.validateAndApply(0xc0005b5280, 0x0, {0x7056a5bb9458, 0xc0005c4b40}, 0xc0002999d0?)
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/vendor/k8s.io/component-base/logs/api/v1/options.go:131 +0x71
k8s.io/component-base/logs/api/v1.ValidateAndApply(...)
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/vendor/k8s.io/component-base/logs/api/v1/options.go:93
k8s.io/kubernetes/cmd/kube-controller-manager/app.NewControllerManagerCommand.func2(0xc00021f808, {0xc0008cf7e0?, 0x0?, 0x1?})
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/cmd/kube-controller-manager/app/controllermanager.go:133 +0x5f
github.com/spf13/cobra.(*Command).execute(0xc00021f808, {0xc00015a010, 0x1, 0x1})
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/vendor/github.com/spf13/cobra/command.go:940 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc00021f808)
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/vendor/github.com/spf13/cobra/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/vendor/github.com/spf13/cobra/command.go:992
k8s.io/component-base/cli.run(0xc00021f808)
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/vendor/k8s.io/component-base/cli/run.go:146 +0x290
k8s.io/component-base/cli.Run(0xc0000061c0?)
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/vendor/k8s.io/component-base/cli/run.go:46 +0x17
main.main()
        /home/fuzzuser/fuzz-projects/kubernetes-1.29.2_kube-controller-manager/project/cmd/kube-controller-manager/controller-manager.go:36 +0x18
```

</details>

--- a/pkg/kubelet/apis/config/validation/validation_test.go
+++ b/pkg/kubelet/apis/config/validation/validation_test.go
@@ -76,7 +76,8 @@ var (
 			"KubeletCrashLoopBackOffMax": true,
 		},
 		Logging: logsapi.LoggingConfiguration{
-			Format: "text",
+			Format:         "text",
+			FlushFrequency: logsapi.TimeOrMetaDuration{Duration: metav1.Duration{Duration: logsapi.LogFlushFreqDefault}},
 		},
 		ContainerRuntimeEndpoint:    "unix:///run/containerd/containerd.sock",
 		ContainerLogMaxWorkers:      1,

--- a/pkg/proxy/apis/config/validation/validation_test.go
+++ b/pkg/proxy/apis/config/validation/validation_test.go
@@ -56,7 +56,8 @@ func TestValidateKubeProxyConfiguration(t *testing.T) {
 			MasqueradeAll: true,
 		},
 		Logging: logsapi.LoggingConfiguration{
-			Format: "text",
+			Format:         "text",
+			FlushFrequency: logsapi.TimeOrMetaDuration{Duration: metav1.Duration{Duration: logsapi.LogFlushFreqDefault}},
 		},
 	}
 	newPath := field.NewPath("KubeProxyConfiguration")
@@ -174,7 +175,8 @@ func TestValidateKubeProxyConfiguration(t *testing.T) {
 		"invalid logging format": {
 			mutateConfigFunc: func(config *kubeproxyconfig.KubeProxyConfiguration) {
 				config.Logging = logsapi.LoggingConfiguration{
-					Format: "unsupported format",
+					Format:         "unsupported format",
+					FlushFrequency: logsapi.TimeOrMetaDuration{Duration: metav1.Duration{Duration: logsapi.LogFlushFreqDefault}},
 				}
 			},
 			expectedErrs: field.ErrorList{field.Invalid(newPath.Child("logging.format"), "unsupported format", "Unsupported log format")},

--- a/staging/src/k8s.io/component-base/logs/api/v1/options.go
+++ b/staging/src/k8s.io/component-base/logs/api/v1/options.go
@@ -138,6 +138,9 @@ func validateAndApply(c *LoggingConfiguration, options *LoggingOptions, featureG
 // can be passed when the struct is not embedded in some larger struct.
 func Validate(c *LoggingConfiguration, featureGate featuregate.FeatureGate, fldPath *field.Path) field.ErrorList {
 	errs := field.ErrorList{}
+	if c.FlushFrequency.Duration.Duration <= 0 {
+		errs = append(errs, field.Invalid(fldPath.Child("flushFrequency"), c.FlushFrequency, "Must be greater than zero"))
+	}
 	if c.Format != DefaultLogFormat {
 		// WordSepNormalizeFunc is just a guess. Commands should use it,
 		// but we cannot know for sure.

--- a/staging/src/k8s.io/component-base/logs/api/v1/validate_test.go
+++ b/staging/src/k8s.io/component-base/logs/api/v1/validate_test.go
@@ -19,9 +19,11 @@ package v1
 import (
 	"math"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
 	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/component-base/featuregate"
 )
@@ -145,6 +147,15 @@ func TestValidation(t *testing.T) {
 			config:      jsonOptionsEnabled,
 			featureGate: enabledFeatureGate,
 		},
+		"Invalid flush frequency": {
+			config: LoggingConfiguration{
+				Format: "text",
+				FlushFrequency: TimeOrMetaDuration{
+					Duration: metav1.Duration{Duration: -1 * time.Second},
+				},
+			},
+			expectErrors: `flushFrequency: Invalid value: -1000000000: Must be greater than zero`,
+		},
 	}
 
 	for name, test := range testcases {
@@ -153,6 +164,7 @@ func TestValidation(t *testing.T) {
 			if featureGate == nil {
 				featureGate = defaultFeatureGate
 			}
+			SetRecommendedLoggingConfiguration(&test.config)
 			err := Validate(&test.config, featureGate, test.path)
 			if len(err) == 0 {
 				if test.expectErrors != "" {

